promql/functions.go

```diff
@@ -524,10 +524,37 @@ func funcLog10(ev *evaluator, args Expressions) model.Value {
 	return vector
 }
 
+// linearRegression performs a least-square linear regression analysis on the
+// provided SamplePairs. It returns the slope, and the intercept value at the
+// provided time.
+func linearRegression(samples []model.SamplePair, interceptTime model.Time) (slope, intercept model.SampleValue) {
```

Comment:
I see it more as an internal helper function that is tested via the `predict_linear` and `deriv` promql tests in `functions.test`. A unit test for this function would be a quite accurate reproduction of https://github.com/prometheus/prometheus/blob/beorn7/storage3/promql/testdata/functions.test#L104-L119 .